### PR TITLE
Add method to use generate release notes endpoint

### PR DESCRIPTION
**Pull request by GuySartorelli**
Adds a method for https://docs.github.com/en/rest/releases/releases#generate-release-notes-content-for-a-release

### CODE REVIEW

File: lib/Github/Api/Repository/Releases.php [hunk 1]
```diff
@@ -68,6 +68,26 @@ public function show($username, $repository, $id)
...
+     *
+     * @throws MissingArgumentException
+     *
+     * @return array
+     */
+    public function generateNotes($username, $repository, array $params)
+    {
+        if (!isset($params['tag_name'])) {
+            throw new MissingArgumentException('tag_name');
+        }
```
Comment (acrobat) [inline lib/Github/Api/Repository/Releases.php:86]: Let's remove this check and fully rely on github to do the validation, if they change anything to the required parameters in the future we don't need a change in the library to be compatible.

**Review by acrobat — CHANGES_REQUESTED** [review]
Thank for the PR @GuySartorelli! Can you check the comment and add a test for the endpoint? Otherwise the PR is good to go! 👌

File: lib/Github/Api/Repository/Releases.php (hunk 1 above)
Comment (GuySartorelli) [reply lib/Github/Api/Repository/Releases.php]: Fair. I was just following what `create()` does in the same class but happy to remove this validation.

**Comment by GuySartorelli** [comment]
Have made the requested change plus added a test.

**Review by acrobat — APPROVED** [review]

**Comment by acrobat** [comment]
Thanks @GuySartorelli! And congrats on your first contribution! 🎉

**Pull request merged**